posthog/api/cohort.py

```diff
@@ -85,14 +88,22 @@ def _handle_csv(self, file, cohort: Cohort) -> None:
     def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> Cohort:
         request = self.context["request"]
         validated_data["created_by"] = request.user
+
         if not validated_data.get("is_static"):
             validated_data["is_calculating"] = True
         cohort = Cohort.objects.create(team_id=self.context["team_id"], **validated_data)
 
         if cohort.is_static:
             self._handle_static(cohort, request)
         else:
-            calculate_cohort_ch.delay(cohort.id)
+
+            with transaction.atomic():
+                cohort = Cohort.objects.filter(pk=cohort.pk).select_for_update().get()
+                cohort.pending_version = cohort.pending_version + 1
+                cohort.save(update_fields=["pending_version"])
+                pending_version = cohort.pending_version
```

Comment:
Might this work, and avoid a transaction:

```suggestion
            cohort.pending_version = F('pending_version') + 1
            cohort.save(update_fields=["pending_version"])
            pending_version = cohort.pending_version
```

as in [here](https://docs.djangoproject.com/en/3.2/ref/models/expressions/#f-expressions)

Reply:
Also, given this is the creation, we can just set this to 0 and leave `version` a null

Reply:
done, i kept the expression for create though because the migration will make version=0 as the default rather than null